src/finch/tensor.py

```diff
@@ -361,6 +364,49 @@ def astype(x: Tensor, dtype: jl.DataType, /, *, copy: bool = True):
         return Tensor(jl.swizzle(result, *x.get_order(zero_indexing=False)))
 
 
+def sum(
+    x: Tensor,
+    /,
+    *,
+    axis: int | tuple[int, ...] | None = None,
+    dtype: np.dtype | None = None,
+    keepdims: bool = False,
+):
+    pass
+
+
+def prod(
+    x: Tensor,
+    /,
+    *,
+    axis: int | tuple[int, ...] | None = None,
+    dtype: np.dtype | None = None,
+    keepdims: bool = False,
+):
+    pass
+
+
+def add(x1: Tensor, x2: Tensor) -> Tensor:
+    pass
+
+
+def multiply(x1: Tensor, x2: Tensor, /) -> Tensor:
+    x1_obj = x1._obj if jl.isa(x1._obj, jl.LogicTensor) else jl.LogicTensor(x1._obj)
+    x2_obj = x2._obj if jl.isa(x2._obj, jl.LogicTensor) else jl.LogicTensor(x2._obj)
```

Comment:
If either is a logic tensor, julia finch should convert the other ones to match.

Reply:
we may want to consider handling "lazy or not lazy" in the julia side, as it's already handled there and the `lazy` boolean doesn't really change how it's implemented in either case. We can keep the boolean, but consider that we don't need to wrap things in lazy or eager here. We also don't need to call compute at the end of this kernel.

Reply:
> If either is a logic tensor, julia finch should convert the other ones to match.

Then converting here should happen only when both are just `SwizzleArrays` (an operation performed on two newly created Tensors)

> We also don't need to call compute at the end of this kernel.

The `compute` is called only for the "eager" mode - when `multiply` is called outside of a `@compile` decorated function. Then how do we know when an operation should be eager or lazy? (i.e. inside or outside of `@compile` decorated function)

Reply:
Or is `LogicTensor` indicating a lazy mode?

Reply:
yes, the `Swizzle` and `Tensor` class are eager, but if any input is `LogicTensor` the operation becomes lazy and produces another LogicTensor

Reply:
I'm hoping to get logictensor running today